re2.asl

```diff
@@ -477,335 +474,25 @@ split
     for (int i = 0; i < currentInventory.Length; i++)
     {
         // Weapon splits
-        if (currentWeapons[i] != oldWeapons[i])
+        if (currentWeapons[i] != oldWeapons[i] && vars.WeaponID.ContainsKey(currentWeapons[i]))
         {
-            switch (currentWeapons[i])
-            {
-                case 0x0000000B:
-                {
-                    if (!vars.Splits.Contains("shotgun"))
-                        return LogAndSplit("shotgun");
-                    break;
-                }
-                case 0x0000002A:
-                {
-                    if (!vars.Splits.Contains("gL"))
-                        return LogAndSplit("gL");
-                    break;
-                }
-                default:
-                {
-                    break; // No work to do.
-                }
-            }
+            string weapon = vars.WeaponID[currentWeapons[i]];
+            if (!vars.Splits.Contains(weapon))
+                return LogAndSplit(weapon);

```

Comment:
We can loop through the item id-name mapping list instead of checking each individually in a switch-case